pyccel/parser/semantic.py

```diff
@@ -1170,12 +1019,12 @@ def _visit_list(self, expr, **settings):
         return Tuple(*ls, sympify=False)
 
     def _visit_tuple(self, expr, **settings):
-        ls = [self._visit(i, **settings) for i in expr]
-        return PythonTuple(ls)
+        ls = tuple(self._visit(i, **settings) for i in expr)
+        return ls
```

Comment:
When do we visit a `tuple` instead of a `PythonTuple` or a Sympy `Tuple`?

Reply:
I'm not sure

Reply:
According to codacy-coverage this function is used, but _visit_Tuple isn't

Reply:
If we don't visit a Sympy `Tuple` anymore, I recommend that we remove the `visit_Tuple` function from our code!

As for my original question, I should rephrase it: where is a Python `tuple` created in our AST? During the syntactic stage or the semantic stage? And is a Python `tuple` the correct node to have there?

Reply:
maybe it's not the correct node and we shouldn't have it, so it must be created somewhere

Reply:
I think this may be coming from the shape. If we translate:
```
from numpy import shape, ones
x = ones((3,4))
s = shape(x)
```
Then we will collect the shape of x which is stored in a `tuple` not a `PythonTuple`

Reply:
I think it's coming from multiple places

Reply:
One of the main places where this is used is when evaluating an If statement. An If is made up of a test and a body these are not stored in test and body structures but in a tuple `(test,body)`

Reply:
@saidctb @EmilyBourne
Thanks to both of you for clarifying this point. I think we can close this conversation now.